lib/Parse/ParseDecl.cpp

```diff
@@ -6656,13 +6656,41 @@ BraceStmt *Parser::parseAbstractFunctionBodyImpl(AbstractFunctionDecl *AFD) {
 
   BraceStmt *BS = Body.get();
   AFD->setBodyParsed(BS);
+  
+  if (BS->getNumElements() == 0)
+    return BS;
+  
+  auto Element = BS->getFirstElement();
```

Comment:
`ParseDecl.cpp` L6663-6731 and `ParseExpr.cpp` L2847-2896 shares  a lot of code. Could you factor them out to a function?

Reply:
Refactored into `Parser::shouldReturnSingleExpressionElement` in Parser.cpp